### PR TITLE
updates app/config/bootstrap scripts

### PR DESCRIPTION
**Pull request by vbresults**
Ping @davidpersson

### CODE REVIEW

**Comment by davidpersson** [comment]
Thanks for the PR. We have to tackle each topic one by one and separate. Each one should have its own commit, they can stay in the same PR. That's OK.

1. cache typo: (safe this will get in) thanks for catching this, it should always be `$next`
2. filters: both ways work, using `::class` looks nicer but has the downside, that we need to load the class file just to apply a filter (even if its not going to be used later). I'll make a decision on this soon. What do you think?
3. Where is `$locale` coming from? The default filter should work fine in most cases, it doesn't try to solve all cases (i.e. I'm personally changing this dependent on project g11n requirements).

File: app/config/bootstrap/cache.php [hunk 1]
```diff
@@ -77,20 +78,20 @@
 	return $result;
 });
 
-Filters::apply('lithium\action\Dispatcher', 'run', function($params, $next) {
+Filters::apply(Dispatcher::class, 'run', function($params, $next) {
 	foreach (Connections::get() as $name) {
 		if (!(($connection = Connections::get($name)) instanceof Database)) {
 			continue;
 		}
 		Filters::apply($connection, 'describe', function($params, $chain) use ($name) {
```
Comment (davidpersson) [inline app/config/bootstrap/cache.php]: $params -> $next

File: app/config/bootstrap/cache.php [hunk 2]
```diff
@@ -77,20 +78,20 @@
...
 			if ($params['fields']) {
-				return $next($params);
+				return $chain($params);
```
Comment (davidpersson) [inline app/config/bootstrap/cache.php]: keep $next

File: app/config/bootstrap/cache.php [hunk 3]
```diff
@@ -77,20 +78,20 @@
...
 			}
 			$cacheKey = "data.connections.{$name}.sources.{$params['entity']}.schema";
 
 			return Cache::read('default', $cacheKey, [
-				'write' => function() use ($params, $next) {
```
Comment (davidpersson) [inline app/config/bootstrap/cache.php]: keep $next

Comment (davidpersson) [inline app/config/bootstrap/cache.php]: keep $next

File: app/config/bootstrap/cache.php (hunk 2 above)
Comment (vbresults) [inline app/config/bootstrap/cache.php]: we can't keep next as it's used by the filter that contains this filter

File: app/config/bootstrap/cache.php (hunk 3 above)
Comment (vbresults) [inline app/config/bootstrap/cache.php]: https://github.com/UnionOfRAD/framework/pull/41#discussion_r71063450

**Comment by vbresults** [comment]
@davidpersson 

1. we can't keep `$next` as this filter is nested within another filter that uses `$next`
2. `::class` doesn't load the class file, which I <3 for setting up filters for cross-dependent plugins (:
```php
// this class/file does not exist
use foo\bar\Baz;

// output: string(11) "foo\bar\Baz"
var_dump(Baz::class);
```
3. `$locale` is coming from a definition earlier in the file (`$locale = 'en';`, I did not add it).  I ran into an intermittent issue when running `li3` commands where the script would kill at the beginning with `Locale `` not found`.

File: app/config/bootstrap/g11n.php [hunk 4]
```diff
@@ -10,16 +10,17 @@
...
+use lithium\console\Dispatcher as ConsoleDispatcher;
 use lithium\core\Environment;
 use lithium\g11n\Locale;
 use lithium\g11n\Catalog;
 use lithium\g11n\Message;
 use lithium\g11n\Multibyte;
+use lithium\net\http\Media;
 use lithium\util\Inflector;
 use lithium\util\Validator;
-use RuntimeException;
```
Comment (davidpersson) [inline app/config/bootstrap/g11n.php]: Please mention the removal of unused dependencies in the commit.

File: app/config/bootstrap/g11n.php [hunk 5]
```diff
@@ -96,6 +97,9 @@
  * @link https://github.com/UnionOfRAD/li3_lldr
  * @link https://github.com/UnionOfRAD/li3_cldr
  */
+
+// use lithium\core\Libraries;
+
```
Comment (davidpersson) [inline app/config/bootstrap/g11n.php]: Seems like a leftover.

File: app/config/bootstrap/g11n.php [hunk 6]
```diff
@@ -10,16 +10,17 @@
  * This bootstrap file contains configurations for all globalizing
  * aspects of your application.
  */
+use lithium\action\Dispatcher as ActionDispatcher;
 use lithium\aop\Filters;
-use lithium\core\Libraries;
```
Comment (davidpersson) [inline app/config/bootstrap/g11n.php]: This might be needed when activating the other catalog config. Let's keep this and remove the commented use before catalog.

**Comment by davidpersson** [comment]
Okay you're right ::class does not do any autoloading:
http://php.net/manual/en/language.oop5.basic.php#language.oop5.basic.class.class

Please separate the usage of ::class into another commit, also we should change the docs in:
https://github.com/UnionOfRAD/lithium/blob/1.1/aop/Filters.php#L88
https://github.com/UnionOfRAD/lithium/blob/1.1/CHANGELOG.md
... to recommend using ":class"

**Comment by vbresults** [comment]
@davidpersson 👍 

Will create a PR for the docs

**Comment by davidpersson** [comment]
Thanks!

File: app/config/bootstrap/cache.php [hunk 7]
```diff
@@ -83,15 +83,15 @@
 		if (!(($connection = Connections::get($name)) instanceof Database)) {
 			continue;
 		}
-		Filters::apply($connection, 'describe', function($params, $chain) use ($name) {
```
Comment (davidpersson) [inline app/config/bootstrap/cache.php]: Actually changing $chain to $next is the only thing needed after looking through this.

Comment (vbresults) [inline app/config/bootstrap/cache.php]: This is done in a separate commit https://github.com/UnionOfRAD/framework/pull/41/commits/cdbc494a9da4fb2c18c4651b8de4fa1e5ec7e05d

**Comment by davidpersson** [comment]
Okay I'll merge this, good? I can do the small changes needed.

**Comment by davidpersson** [comment]
Merged in 
94c4cffdca6a73ba074db04d779a5c887da8ffcc
4c3e19dc5322f24f31890127592deda0550152a5
